streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNode.java

```diff
@@ -29,14 +30,20 @@
     private ProcessorContext context;
     private Deserializer<K> keyDeserializer;
     private Deserializer<V> valDeserializer;
+    private final TimestampExtractor timestampExtractor;
 
-    public SourceNode(String name, List<String> topics, Deserializer<K> keyDeserializer, Deserializer<V> valDeserializer) {
+    public SourceNode(String name, List<String> topics, TimestampExtractor timestampExtractor, Deserializer<K> keyDeserializer, Deserializer<V> valDeserializer) {
```

Comment:
Yes, once the source is defined and added to topology, we cannot tell if it is defined by name or by pattern. This was the case before this PR as well.